sport-moment-nft/contracts/EnglishPremierLeague.cdc

```diff
@@ -0,0 +1,843 @@
+
+    /// Named Paths
+    ///
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let AdminStoragePath:       StoragePath
+    pub let MinterPrivatePath:      PrivatePath
+
+    /// Contract variables
+    ///
+    pub var royaltyAddress: Address
+    pub var totalSupply:        UInt64
+    pub var nextSeriesID:       UInt64
+    pub var nextSetID:          UInt64
+    pub var nextTagID:          UInt64
+    pub var nextPlayID:         UInt64
+    pub var nextEditionID:      UInt64
+
+    /// Metadata Dictionaries
+    ///
+    access(self) let seriesIDByName:    {String: UInt64}
+    access(self) let seriesByID:        {UInt64: Series}
+    access(self) let setIDByName:       {String: UInt64}
+    access(self) let setByID:           {UInt64: Set}
+    access(self) let tagByID:           {UInt64: Tag}
+    access(self) let playByID:          {UInt64: Play}
+    access(self) let editionByID:       {UInt64: Edition}
+
+    /// A top-level Series with a unique ID and name
+    ///
+    pub struct Series {
+        pub let id: UInt64
+        pub let name: String
+        pub var active: Bool
+
+        init (id: UInt64, name: String) {
+            if let series = EnglishPremierLeague.seriesByID[id] {
+                self.id = series.id
+                self.name = series.name
+                self.active = series.active
+            } else {
+                self.id = id
+                self.name = name!
+                self.active = true
+            }
+        }
+
+        pub fun close() { self.active = false }
+    }
+
+    /// Series getters
+    ///
+    pub fun getSeries(id: UInt64): EnglishPremierLeague.Series {
+        return EnglishPremierLeague.seriesByID[id]!
+    }
+
+    pub fun getSeriesByName(name: String): EnglishPremierLeague.Series? {
+        let id = EnglishPremierLeague.seriesIDByName[name]
+
+        if id == nil{
+            return nil
+        }
+
+        return EnglishPremierLeague.getSeries(id: id!)
+    }
+
+    pub fun getAllSeriesNames(): [String] {
+        return EnglishPremierLeague.seriesIDByName.keys
+    }
+
+    pub fun getSeriesIDByName(name: String): UInt64? {
+        return EnglishPremierLeague.seriesIDByName[name]
+    }
+
+    /// A top-level Set with ID and name
+    ///
+    pub struct Set {
+        pub let id: UInt64
+        pub let name: String
+        pub var locked: Bool
+
+        init (id: UInt64, name: String) {
+            if let set = EnglishPremierLeague.setByID[id] {
+                self.id = set.id
+                self.name = set.name
+                self.locked = set.locked
+            } else {
+                self.id = id
+                self.name = name!
+                self.locked = false
+            }
+        }
+
+        pub fun lock() { self.locked = true }
+    }
+
+    /// Set getters
+    ///
+    pub fun getSet(id: UInt64): EnglishPremierLeague.Set {
+        return EnglishPremierLeague.setByID[id]!
```

Comment:
Again same better to return the optional value

Reply:
Done. Thank you.